### PR TITLE
Gi 158 error screen

### PR DESCRIPTION
**Pull request by MerlindCPohl**
## Description

implements three difference error screens:
401
calling websites unauthorized, f.e. /settings while being logged out

404
appears when website doesn't exist f.e. /quark

500 
appears when API fetch was not successful, implemented in all api calling pages.
reloading the page is possible with the variable intendedDestination, so the user can actively try to reload the page

## Jira Ticket

Link to the Jira ticket: [[GI-XXX](https://techstudyfinder.atlassian.net/browse/GI-XXX)](https://techstudyfinder.atlassian.net/jira/software/projects/GI/boards/1?selectedIssue=GI-158)

## Author Checklist

- [x ] Code follows project coding standards and conventions
- [x ] Tests have been written or updated
- [x ] All tests pass locally
- [x ] No sensitive data or secrets are included
- [x ] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

## Additional Context (optional)

Add any background information or references that help reviewers understand the changes.

---

_Please ensure all criteria are met before merging._

### CODE REVIEW

**Comment by MerlindCPohl** [comment]
Bildschirmfoto 2026-01-15 um 17 25 15
Bildschirmfoto 2026-01-15 um 17 25 50
Bildschirmfoto 2026-01-15 um 18 04 30

File: client/src/api/authApi.ts [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+  fetchFn: typeof fetch = fetch,
+) {
+  const res = await fetchFn("/api/auth/login", {
     method: "POST",
     credentials: "include",
     headers: { "Content-Type": "application/json" },
     body: JSON.stringify({ username, password }),
   });
-  if (!res.ok) throw new Error("Login failed");
+  if (!res.ok) throw new Error("Login fehlgeschlagen");
```
Comment (psophis) [inline client/src/api/authApi.ts]: Sollte englisch bleiben. Die Projektsprache ist englisch und der geworfene Text wird nicht in der UI ausgegeben.

File: client/src/theme/theme.ts [hunk 2]
```diff
@@ -52,6 +52,12 @@ const theme = createTheme({
   },
   typography: {
     fontFamily: "Fira Sans, sans-serif",
+    h1: {
+      fontFamily: '"Lexend Deca", Fira Sans, sans-serif',
+      fontSize: "6rem", // 48px on desktop
+      fontWeight: "bold",
+      marginBottom: "2000px",
```
[the remark this answers is not in the document]
Comment (MerlindCPohl) [reply client/src/theme/theme.ts]: Hab eine customize headline angelegt: errorScreenTitle :)
